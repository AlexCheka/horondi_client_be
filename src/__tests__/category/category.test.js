/* eslint-disable no-undef */
const { gql } = require('apollo-boost');
const categoryService = require('../../modules/category/category.service')
const categoryResolver = require('../../modules/category/category.resolver')
const client = require('../../utils/apolloClient');

require('dotenv').config();


describe('querries', () => {
    test('should recive all categories like in service and resolver', async () => {
        const res = await client.query({
            query: gql`
                query {       
                    getAllCategories {
                        available
                        name {
                            value
                            lang
                        }
                        categoryCode
                        images {
                            large
                            medium
                            small
                            thumbnail
                        }
                    }
                }
            `
        }).then(res => res)

        expect(categoryService.getAllCategories()).resolves.toBe(res);

        expect(res.data.getAllCategories).toMatchSnapshot()
    })

    test('should recive category by id like in service and resolver', async () => {
      const res = await client.query({
          query: gql`
              query {       
                getCategoryById(id: "5ef3970c0ab5dd42436dd5c4"){
                  _id
                  name {
                    value
                    lang
                  }
                  categoryCode
                  images {
                    large
                    medium
                    small
                    thumbnail
                  }
                }
              }
          `
      }).then(res => res)

      expect(categoryService.getCategoryById("5ef3970c0ab5dd42436dd5c4")).resolves.toBe(res);
      expect(categoryResolver.categoryQuery.getCategoryById('',{id:"5ef3970c0ab5dd42436dd5c4"})).resolves.toBe(res);
      expect(res.data.getCategoryById).toMatchSnapshot()
  })
})