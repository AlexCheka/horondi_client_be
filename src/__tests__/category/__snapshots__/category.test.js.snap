// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`querries should recive all categories like in service and resolver 1`] = `
Array [
  Object {
    "available": true,
    "categoryCode": "main-backpacks",
    "images": Object {
      "large": "https://horondi.blob.core.windows.net/horondi/categories/backpacks.jpg",
      "medium": null,
      "small": null,
      "thumbnail": null,
    },
    "name": Array [
      Object {
        "lang": "uk",
        "value": "Рюкзаки",
      },
      Object {
        "lang": "en",
        "value": "Backpacks",
      },
    ],
  },
  Object {
    "available": true,
    "categoryCode": "main-bags",
    "images": Object {
      "large": "https://horondi.blob.core.windows.net/horondi/categories/bags.jpg",
      "medium": null,
      "small": null,
      "thumbnail": null,
    },
    "name": Array [
      Object {
        "lang": "uk",
        "value": "Сумки",
      },
      Object {
        "lang": "en",
        "value": "Bags",
      },
    ],
  },
  Object {
    "available": true,
    "categoryCode": "main-accessories",
    "images": Object {
      "large": "https://horondi.blob.core.windows.net/horondi/categories/accessories.jpg",
      "medium": null,
      "small": null,
      "thumbnail": null,
    },
    "name": Array [
      Object {
        "lang": "uk",
        "value": "Аксесуари",
      },
      Object {
        "lang": "en",
        "value": "Accessories",
      },
    ],
  },
]
`;

exports[`querries should recive category by id like in service and resolver 1`] = `
Object {
  "_id": "5ef3970c0ab5dd42436dd5c4",
  "categoryCode": "main-backpacks",
  "images": Object {
    "large": "https://horondi.blob.core.windows.net/horondi/categories/backpacks.jpg",
    "medium": null,
    "small": null,
    "thumbnail": null,
  },
  "name": Array [
    Object {
      "lang": "uk",
      "value": "Рюкзаки",
    },
    Object {
      "lang": "en",
      "value": "Backpacks",
    },
  ],
}
`;
